src/Bicep.LangServer.IntegrationTests/Helpers/IntegrationTestHelper.cs

```diff
@@ -22,7 +23,7 @@ public static async Task<ILanguageClient> StartServerWithClientConnection(Action
             var clientPipe = new Pipe();
             var serverPipe = new Pipe();
 
-            var server = new Server(serverPipe.Reader, clientPipe.Writer);
+            var server = new Server(TestResourceTypeProvider.CreateRegistrar(), serverPipe.Reader, clientPipe.Writer);
```

Comment:
>TestResourceTypeProvider.CreateRegistrar() [](start = 36, length = 42)

I think to be able to take advantage of DI in the server, we should expose a callback to override the container contents in tests. I wouldn't expect integration tests to do it, but would be useful for unit tests probably.

Reply:
Done, and I also removed `ResourceTypeRegistrar` entirely; it's just a wrapper around `IResourceTypeProvider` at this point. We can add it back at some point if we ever have the need to support multiple `IResourceTypeProvider` instances or dynamic registrations of instances.